public/main.js

```diff
@@ -17,26 +18,28 @@ var messages = [
 var ALL_VALUES = 0;
 var ACTIVE_VALUES = 1;
 var COMPLETE_VALUES = 2;
-var mode = ALL_VALUES;
-displayMessage.innerHTML = messages[ALL_VALUES];
+var mode;
+//this refreshes the page and initialises display mode to show all items in list
+modeSetter(ALL_VALUES);
 
+//---------------------Actual-functionality-------------------------------------
 all.onclick = function() {
-    mode = ALL_VALUES;
-    displayMessage.innerHTML = messages[mode];
-    reloadTodoList();
+    modeSetter(ALL_VALUES);
 };
 
 active.onclick = function() {
-    mode = ACTIVE_VALUES;
-    displayMessage.innerHTML = messages[mode];
-    reloadTodoList();
+    modeSetter(ACTIVE_VALUES);
 };
 
 completed.onclick = function() {
-    mode = COMPLETE_VALUES;
+    modeSetter(COMPLETE_VALUES);
+};
+
+function modeSetter(newMode) {
```

Comment:
It's probably better to call it `setMode`. Otherwise it's not entirely obvious what is happening.